src/Providers/components/ManageColumnsToolbar.tsx

```diff
@@ -0,0 +1,182 @@
+import React, { useState } from 'react';
+import { useTranslation } from 'src/internal/i18n';
+
+import {
+  Button,
+  DataList,
+  DataListCell,
+  DataListCheck,
+  DataListControl,
+  DataListDragButton,
+  DataListItem,
+  DataListItemCells,
+  DataListItemRow,
+  DragDrop,
+  Draggable,
+  Droppable,
+  Modal,
+  Text,
+  TextContent,
+  TextVariants,
+  ToolbarItem,
+  Tooltip,
+} from '@patternfly/react-core';
+import { ColumnsIcon } from '@patternfly/react-icons';
+
+import { Field } from './shared';
+
+export const ManageColumnsToolbar = ({
+  fields,
+  setFields,
+  defaultFields,
+}: {
+  fields: Field[];
+  defaultFields: Field[];
+  setFields(fileds: Field[]): void;
+}) => {
+  const { t } = useTranslation();
+  const [manageColumns, setManageColumns] = useState(false);
+  return (
+    <ToolbarItem>
+      <Tooltip content={t('plugin__forklift-console-plugin~ManageColumns')}>
+        <Button
+          variant="plain"
+          onClick={() => setManageColumns(true)}
+          aria-label={t('plugin__forklift-console-plugin~ManageColumns')}
+        >
+          <ColumnsIcon />
+        </Button>
+      </Tooltip>
+      <ManageColumns
+        showModal={manageColumns}
+        onClose={() => setManageColumns(false)}
+        description="Selected columns will be displayed in the table."
+        columns={fields}
+        onChange={setFields}
+        defaultColumns={defaultFields}
+      />
+    </ToolbarItem>
+  );
+};
+
+interface ManagedColumnsProps {
+  showModal: boolean;
+  description: string;
+  onClose(): void;
+  onChange(colums: Field[]): void;
+  columns: Field[];
+  defaultColumns: Field[];
+}
+
+const ManageColumns = ({
+  showModal,
+  description,
+  onClose,
+  onChange,
+  columns,
+  defaultColumns,
+}: ManagedColumnsProps) => {
+  const { t } = useTranslation();
+  const [editedColumns, setEditedColumns] = useState(columns);
```

Comment:
we should save the preferred columns on localStorage

Reply:
We can add this in follow up.